main.go

```diff
@@ -0,0 +1,333 @@
+// Web Chat in Go is a simple chat built with the websocket protocol (rfc6455)
+// on the back end and a free bootstrap template on the front end.
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"time"
+
+	"web-chat-go/hub"
+	"web-chat-go/user"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+	_ "github.com/lib/pq"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type session struct {
+	un           string
+	lastActivity time.Time
+}
+
+var tpl *template.Template
+var upgrader = websocket.Upgrader{
+	// BufferSize determines how many bytes does the CPU handles each load,
+	// the bigger the size the less the CPU is going to work as less loads
+	// would need to be processed.
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// UserName, User
+var dbUsers = make(map[string]user.User)
+
+// Session ID, UserName
+var dbSessions = make(map[string]session)
+
+// The single instance of "hub" is assigned.
+var h = hub.NewHub()
+
+var tcpConn = make(chan hub.Client)
+
+func init() {
+	tpl = template.Must(template.ParseGlob("./assets/html/*.gohtml"))
+}
+
+func main() {
+	go h.Run()
+
+	http.HandleFunc("/", index)
+	http.HandleFunc("/signup/", signUp)
+	http.HandleFunc("/login/", login)
+	http.HandleFunc("/logout/", logout)
+	http.HandleFunc("/ws/", serveWs)
+
+	http.Handle("/assets/", http.StripPrefix("/assets",
+		http.FileServer(http.Dir("./assets"))))
+
+	go func() {
+		// A listener is opened in the port "6893" with the "tcp" network.
+		li, err := net.Listen("tcp", ":6893")
+		if err != nil {
+			panic(err)
+		}
+		defer li.Close()
+
+		// Infinitely accepts for new connections and sends a goroutine for
+		// each one that handles it.
+		for {
+			conn, err := li.Accept()
+			if err != nil {
+				panic(err)
+			}
+
+			go handle(conn)
+		}
+	}()
+
+	err := http.ListenAndServe(":80", nil)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// handle takes the accepted connection and asks for an identification
+// from the user. After a scanner brought by the "bufio" package reads
+// each line inputted by the user and prints them to the server and
+// any other connection instantiated hub, hundle will close the connection.
+func handle(conn net.Conn) {
```

Comment:
handle is an overloaded term in this app already, so you should choose a longer name